src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -461,28 +459,52 @@ private EntityPlayerMP parseAnyPlayerValue(@NotNull final MinecraftServer server
     private IColony parseColonyValue(@NotNull final ICommandSender sender, @NotNull final MineColonyDataProvider mineColonyDataProvider, final String potentialArgumentValue)
     {
         final List<String> colonyNumberStrings = getColonyIdStrings(mineColonyDataProvider);
-        final Integer result = Ints.tryParse(potentialArgumentValue);
-        if (null != result)
-        {
-            int colonyNumber = result.intValue();
-            if (sender instanceof EntityPlayer && colonyNumber == -1)
-            {
-                final IColony icolony = mineColonyDataProvider.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
-                if (icolony != null)
-                {
-                    colonyNumber = icolony.getID();
-                }
-            }
-            if (colonyNumberStrings.contains(String.valueOf(colonyNumber)))
-            {
-                return mineColonyDataProvider.getColony(colonyNumber, sender.getEntityWorld().provider.getDimension());
-            }
-            return null;
-        }
-        else
-        {
-            return null;
-        }
+        int dimensionId;
+        int colonyNumber;
+        if (potentialArgumentValue.contains("|"))
+        {
+        	String[] split = potentialArgumentValue.split(Pattern.quote("|"));
+        	dimensionId = Ints.tryParse(split[0]).intValue();
+        	colonyNumber = Ints.tryParse(split[1]).intValue();
+        }
+        else if (sender instanceof EntityPlayer)
+        {
+        	final Integer result = Ints.tryParse(potentialArgumentValue);
+        	if (null != result) {
+        		dimensionId = ((EntityPlayer) sender).dimension;
+        		colonyNumber = result.intValue();
+        	}
+        	else return null;
+        }
+        else 
+        {
+        	return null;
+        }
+    	if (sender instanceof EntityPlayer && colonyNumber == -1) 
+    	{
+    		final IColony icolony = mineColonyDataProvider.getIColonyByOwner(FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(dimensionId), (EntityPlayer) sender);
+    		if (icolony != null)
+    		{
+    			colonyNumber = icolony.getID();
+    		}
+    	}
+    	if (colonyNumberStrings.contains(dimensionId + "|" + colonyNumber))
```

Comment:
earlier we check for the potentialArgumentValue and now for colonyNumberStrings? Why check in different places?

Reply:
That is in case we assume the dimension ID

Reply:
kk

Reply:
Oh I see what you misunderstood. I'm checking whether the colony in that dimension exists.

Reply:
That has nothing to do with the potentialArgumentValue containing the pipe character